// @quasar/quasar-app-extension-testing-unit-jest/dist/layout-injections.js
"use strict";
// TODO: we need the injection keys to be typed and exported from Quasar package
Object.defineProperty(exports, "__esModule", { value: true });
exports.qLayoutInjections = void 0;
const globals_1 = require("@jest/globals");
const vue_1 = require("vue");
/**
 * Injections for Components with a QPage root Element
 */
function qLayoutInjections() {
    return {
        // pageContainerKey
        _q_pc_: true,
        // layoutKey
        _q_l_: {
            header: { size: 0, offset: 0, space: false },
            right: { size: 300, offset: 0, space: false },
            footer: { size: 0, offset: 0, space: false },
            left: { size: 300, offset: 0, space: false },
            isContainer: (0, vue_1.ref)(false),
            view: (0, vue_1.ref)('lHh Lpr lff'),
            rows: (0, vue_1.ref)({ top: 'lHh', middle: 'Lpr', bottom: 'lff' }),
            height: (0, vue_1.ref)(900),
            instances: {},
            update: globals_1.jest.fn(),
            animate: globals_1.jest.fn(),
            totalWidth: (0, vue_1.ref)(1200),
            scroll: (0, vue_1.ref)({ position: 0, direction: 'up' }),
            scrollbarWidth: (0, vue_1.ref)(125),
        },
    };
}
exports.qLayoutInjections = qLayoutInjections;
